### ISSUE TITLE
Deadlock in --enable-locks build

### ISSUE THREAD

**Issue by p-steuer** [opencryptoki/opencryptoki#146]
When build with the --enable-locks configure option, a deadlock occurs in the pkcs11test suite's ParallelSessions test.

edit:
lock is in C_CloseAllSessions

```
(gdb) bt
#0  0x000003fffde0c7be in pthread_rwlock_wrlock () from /lib64/libpthread.so.0
#1  0x000003fffdf0fb30 in bt_destroy () from /usr/lib/opencryptoki/libopencryptoki.so
#2  0x000003fffdf0e81c in CloseAllSessions () from /usr/lib/opencryptoki/libopencryptoki.so
#3  0x000003fffdf0d460 in C_CloseAllSessions () from /usr/lib/opencryptoki/libopencryptoki.so
#4  0x0000000001047050 in pkcs11::test::PKCS11Test_ParallelSessions_Test::TestBody (this=0x1238350) at session.cc:137
#5  0x00000000010f0a9a in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void> (object=0x1238350, method=&virtual testing::Test::TestBody(), 
    location=0x1121af8 "the test body") at gtest-1.6.0/src/gtest.cc:2090
#6  0x00000000010e9d06 in testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void> (object=0x1238350, method=&virtual testing::Test::TestBody(), 
    location=0x1121af8 "the test body") at gtest-1.6.0/src/gtest.cc:2126
#7  0x00000000010d01aa in testing::Test::Run (this=0x1238350) at gtest-1.6.0/src/gtest.cc:2161
#8  0x00000000010d0b50 in testing::TestInfo::Run (this=0x119ed10) at gtest-1.6.0/src/gtest.cc:2338
#9  0x00000000010d12e4 in testing::TestCase::Run (this=0x119c680) at gtest-1.6.0/src/gtest.cc:2445
#10 0x00000000010d7a06 in testing::internal::UnitTestImpl::RunAllTests (this=0x11975c0) at gtest-1.6.0/src/gtest.cc:4237
#11 0x00000000010f241a in testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (object=0x11975c0, 
    method=(bool (testing::internal::UnitTestImpl::*)(testing::internal::UnitTestImpl * const)) 0x10d7660 <testing::internal::UnitTestImpl::RunAllTests()>, 
    location=0x1122618 "auxiliary test code (environments or event listeners)") at gtest-1.6.0/src/gtest.cc:2090
#12 0x00000000010eb1d6 in testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (object=0x11975c0, 
    method=(bool (testing::internal::UnitTestImpl::*)(testing::internal::UnitTestImpl * const)) 0x10d7660 <testing::internal::UnitTestImpl::RunAllTests()>, 
    location=0x1122618 "auxiliary test code (environments or event listeners)") at gtest-1.6.0/src/gtest.cc:2126
#13 0x00000000010d6384 in testing::UnitTest::Run (this=0x1184378 <testing::UnitTest::GetInstance()::instance>) at gtest-1.6.0/src/gtest.cc:3871
#14 0x000000000100717c in main (argc=12, argv=0x3fffffff2e8) at pkcs11test.cc:249
```

**Comment by p-steuer** [comment]
Fixed by https://github.com/opencryptoki/opencryptoki/pull/142/commits/244260a2159891bfc50ecfc3aa28a538cfd0b712

**Comment by p-steuer** [comment]
Fixed by PR #142. Closing.
